vignettes/trial_sanity_checks.Rmd

```diff
@@ -0,0 +1,338 @@
+next_dose <- maxDose(increment_rule, no_tox_below_25_data)
+next_dose
+```
+
+
+Thus, the initial dose rule permanently prevents escalation above 25 mg because the highest permitted dose above 25 mg is `r next_dose` but the lowest such dose in the dose grid is 40.  The problem can be solved in at least two ways: we can either relax the increments rule or we can introduce one or more intermediate doses.  (We could, of course, do both.)  We choose to relax the increments rule and check that, ceteris paribus, this permits escalation from every dose:
+
+```{r}
+revised_increment_rule <- IncrementsRelative(
+  intervals = c(0, 20, 80),
+  increments = c(1, 0.67, 0.33)
+)
+tibble(
+  HighestDoseUsed = dose_grid,
+  MaxPermittedDose = c(
+    NA,
+    sapply(
+      1:length(dose_grid[-1]),
+      function(n) {
+        tmp <- dose_grid[-1]
+        d <- Data(
+          doseGrid = dose_grid,
+          x = tmp[1:n],
+          y = rep(0, n),
+          ID = as.integer(1:n),
+          cohort = as.integer(1:n)
+        )
+        maxDose(revised_increment_rule, d)
+      }
+    )
+  )
+) %>%
+  mutate(EscalationPermitted = lead(HighestDoseUsed < MaxPermittedDose)) %>%
+  kable()
+```
+
+Yes, it does.  So we update the design:
+```{r}
+revised_design <- Design(
+  model = initial_model,
+  nextBest = next_best,
+  stopping = stopping_trial,
+  increments = revised_increment_rule,
+  cohortSize = cohort_size,
+  data = empty_data,
+  startingDose = 3
+)
+examine(revised_design) %>%
+  kable()
+```
+
+Hmmm. We now appear to be stuck at 50 mg.  Why is this?  Let's examine the state of the model after escalation to 50 mg without toxicity...
+
+```{r}
+no_tox_below_50_data <- Data(
+  doseGrid = dose_grid,
+  x = c(1, 3, 5, 10, 15, 20, 25, 40, 40, 40, 50, 50, 50),
+  y = rep(0, 13),
+  ID = 1L:13L,
+  cohort = c(1L:7L, rep(8L:9L, each = 3))
+)
+
+default_mcmc_options <- McmcOptions(burnin = 1000, step = 2, samples = 1000)
+no_tox_50_samples <- mcmc(no_tox_below_50_data, initial_model, default_mcmc_options)
+recommended_dose <- nextBest(
+  next_best,
+  doselimit = Inf,
+  samples = no_tox_50_samples,
+  model = initial_model,
+  data = no_tox_below_50_data
+)
+recommended_dose$value
+recommended_dose$probs
+
+maxDose(revised_increment_rule, no_tox_below_50_data)
+```
+
+Whilst the increments rule allows escalation to 83.5 mg, the toxicity estimates provided by the model do not: the current estimate of toxicity at 80 mg is `r recommended_dose$probs[10, 3]`, well above the limit of 0.25 defined in the dose recommendation rule (`next_best`).
+
+Again, we have several options.  We can introduce intermediate doses, thus slowing escalation and allowing the prior model more time to adapt (but probably requiring a larger trial), or we can modify the prior so that it allows faster, but still reasonable, escalation.  Let's try creating a minimally informative prior that is consistent with our first attempt.
+
+```{r, error=TRUE}
+# Fitting the min-inf_model is slow.
```

Comment:
it would be good for this chunk to avoid commented code and instead work with an if/else condition - if the save file is there, skip the code execution and load it, otherwise execute the code and save to the file.